bundles/org.openhab.binding.myenergi/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,453 @@
+				<option value="3">Eco+</option>
+				<option value="4">Stop</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="zappi-status-type">
+		<item-type>String</item-type>
+		<label>Status</label>
+		<description>The current status of the device</description>
+		<state readOnly="true">
+			<options>
+				<option value="0">Starting</option>
+				<option value="1">Waiting for Export</option>
+				<option value="2">Demand Side Response</option>
+				<option value="3">Diverting</option>
+				<option value="4">Boosting</option>
+				<option value="5">Charge Complete</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="plug-status-type">
+		<item-type>String</item-type>
+		<label>Plug Status</label>
+		<description>The current plug status of the device</description>
+		<state readOnly="true">
+			<options>
+				<option value="A">EV Disconnected</option>
+				<option value="B1">Awaiting Surplus</option>
+				<option value="B2">Charge Complete</option>
+				<option value="C1">Transitory - unknown</option>
+				<option value="C2">Charging Max Power</option>
+				<option value="F">Fault/Restart</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="eddi-status-type">
+		<item-type>String</item-type>
+		<label>Status</label>
+		<description>The current status of the device</description>
+		<state readOnly="true">
+			<options>
+				<option value="0">Starting</option>
+				<option value="1">Awaiting Surplus</option>
+				<option value="3">Diverting</option>
+				<option value="4">Boosting</option>
+				<option value="5">Max temp reached</option>
+				<option value="6">Stopped</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="command-tries-type">
+		<item-type>Number</item-type>
+		<label>Command Status/Retries</label>
+		<description>0-10 Trying, 253 Acked and Failed, 254 Acked and OK, 255 No command has ever been sent</description>
+		<state readOnly="true" pattern="%d"/>
+	</channel-type>
+
+	<channel-type id="diverter-priority-command">
+		<item-type>Number</item-type>
+		<label>Diverter Priority</label>
+		<description>The diverter priority</description>
+		<state pattern="%d" min="1" max="10" step="1"/>
+	</channel-type>
+
+	<channel-type id="diverter-priority-type">
+		<item-type>Number</item-type>
+		<label>Diverter Priority</label>
+		<description>The diverter priority</description>
+		<state pattern="%d"/>
+	</channel-type>
+
+	<channel-type id="heater-priority-type">
+		<item-type>Number</item-type>
+		<label>Heater Priority</label>
+		<description>The priority heater</description>
+		<state readOnly="false" pattern="%d"/>
+	</channel-type>
+
+	<channel-type id="minimum-green-level-type">
+		<item-type>Number</item-type>
+		<label>Minimum Green Level</label>
+		<description>The minimum percentage of green energy to allow charging</description>
+		<state readOnly="true" pattern="%d %%"/>
+	</channel-type>
+
+	<channel-type id="boost-end-time-type">
```

Comment:
Is this a timestamp or a remaining duration? Either wya, better to make use of UoM